apps/dashboard/app/models/batch_connect/app.rb

```diff
@@ -297,5 +302,23 @@ def submit_config(binding: nil)
       def ood_app
         @ood_app ||= OodApp.new(router)
       end
+
+      # add a widget for choosing the cluster.
+      def add_cluster_widget(attributes, attribute_list)
+        attribute_list.prepend("cluster") unless attribute_list.include? "cluster"
+
+        if clusters.size > 1
+          attributes[:cluster] = {
+            widget: "select",
+            label: "Cluster",
+            options: clusters.map { |cluster| cluster.id.to_s }
+          }
+        else
+          attributes[:cluster] = {
+            value: clusters.first.id.to_s,
+            widget: "hidden_field"
+          }
+        end
+      end
```

Comment:
How do we handle the situation where we want to define our own dropdown or set of dropdowns that eventually set cluster?

Reply:
Yea I thought it'd be additive, like we'd build on this. As an example, we'd want to let add new options text, but have logic so they can't sneak in arbitrary clusters.  More than that, we have to enforce they're using a 'select' widget.

I just figured this is the safest initial option that we can build on later.

Reply:
>logic so they can't sneak in arbitrary clusters

I think that client side, the view ultimately should be able to specify whatever client is desired (because in the end that is what the user can do if manipulating the form in browser dev tools). Server side we should have some validation that whatever was submitted client side is legitimate.